src/Microsoft.VisualStudio.Composition/ByValueEquality+AssemblyNameComparer.cs

```diff
@@ -32,8 +40,8 @@ public bool Equals(AssemblyName x, AssemblyName y)
                     return true;
                 }
 
-                // fast path
-                if (x.CodeBase == y.CodeBase)
+                // If fast check is enabled, we can compare the code bases
+                if (this.fastCheck && x.CodeBase == y.CodeBase)
```

Comment:
Who still depends on the 'fast' check? Do they really need it to be fast, or can we simplify by removing this fast check altogether?